src/testing/starknet_state.rs

```diff
@@ -196,7 +197,8 @@ impl StarknetState {
         self.state = self.state.apply_to_copy();
         let tx = tx.execute(&mut self.state, &self.general_config)?;
         let tx_execution_info = ExecutionInfo::Transaction(Box::new(tx.clone()));
-        self.add_messages_and_events(&tx_execution_info);
+        self.add_messages_and_events(&tx_execution_info)
+            .map_err::<TransactionError, _>(|_| todo!())?;
```

Comment:
I don't get why you make changes to this file.
Also, add an error and remove this todo!()

Reply:
The `todo!()` was probably there because I couldn't figure which error to return, or ended up with a recursive error trying to fix it.

As for the changes, I had to add the `.map_err()` after I got an unused result error after removing an `#[allow(unused)]`.